lib/templates-sync.php

```diff
@@ -18,20 +18,21 @@
  * @param string $content   Template content.
  */
 function _gutenberg_create_auto_draft_for_template( $post_type, $slug, $theme, $content ) {
-	// We check if an auto-draft was already created,
+	// We check if a post for this template was already created,
 	// before running the REST API calls
-	// because the site editor needs an existing auto-draft
+	// because the site editor needs an existing post
 	// for each theme template part to work properly.
 	$template_query = new WP_Query(
 		array(
 			'post_type'      => $post_type,
-			'post_status'    => array( 'publish', 'auto-draft' ),
+			'post_status'    => array( 'publish' ),
 			'post_name__in'  => array( $slug ),
 			'tax_query'      => array(
 				array(
 					'taxonomy' => 'wp_theme',
 					'field'    => 'slug',
-					'terms'    => $theme,
+					'terms'    => array( $theme, '_wp_is_original' ),
+					'operator' => 'AND',
```

Comment:
Is this needed? If not, is there merit to changing these two "special" terms to post meta instead, since it won't be necessary to query them? This also means they won't conflict with theme names.

Reply:
I think the chance for a conflict with the theme name is very close to zero. We decided to use terms to avoid post meta perf problems.

Reply:
I thought the performance problems only apply when you're querying posts by meta, rather than querying meta by post. What I don't understand is why it needs to query posts by `_wp_is_original`, shouldn't it only be querying by the theme name?

Reply:
You are right, the query is wrong here. This is always going to create a new `file-based` `wp_template` whenever it can't find an original `wp_template`. We don't need to query by `_wp_is_original` here.

I'm not sure if we are ever going to query by `_wp_is_original` or `_wp_is_file_based`. So your perf observations are right. For the sake of simpler code and queries (and future-proofing), I'd keep these two special values in the `theme` taxonomy. `postmeta` table is always huge compared to the `terms` table, so in the long run it _should_ be faster. I'm looking forward to others' opinion too.

Reply:
I would also say that if they are kept as taxonomy terms, they probably shouldn't be in the same taxonomy as the theme names, because it would make it more complicated to count how many themes a template belongs to.

Reply:
Right, we also discussed introducing a generic `tags` taxonomy for all the current and upcoming flags.